assert/string.go

```diff
@@ -0,0 +1,37 @@
+package assert
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+)
+
+// StringContains asserts that string s contains the subst string and calls
+// the failure function with details otherwise.
+func (assert *Assert) StringContains(s string, substr string, details ...interface{}) {
+	assert.IsTrue(strings.Contains(s, substr), "strings.Contains(%q, %q).%s",
+		s, substr, errordetails(details...))
+}
+
+// StringMatch asserts that string matches the regex pattern and calls
+// the failure function with details otherwise.
+func (assert *Assert) StringMatch(pattern string, str string, details ...interface{}) {
+	matched, err := regexp.MatchString(pattern, str)
+	assert.IsFalse(err != nil, "err != nil. %v", pattern, err)
```

Comment:
Don't we have assert.Error/assert.NoError ? they are usually quite useful when you are not interested on the specifics of errors.